rules/external-secret-storage/raw.rego

```diff
@@ -0,0 +1,54 @@
+package armo_builtins
+
+import future.keywords.every
+
+# Encryption config is not using a recommended provider for KMS
+deny[msg] {
+	obj = input[_]
+	is_control_plane_info(obj)
+	config_file := obj.data.APIServerInfo.encryptionProviderConfigFile
+	config_file_content = decode_config_file(base64.decode(config_file.content))
+
+	resources := config_file_content.resources
+	resource := resources[_]
+	every resource in resources{
+		not has_recommended_provider(resource)
+	}
+
+	fix_paths := [
+	{"path": sprintf("resources[%d].resources[%d]", [count(resources), 0]),	"value": "secrets"},
+	{"path": sprintf("resources[%d].providers[%d].kms", [count(resources), 0]),	"value": "YOUR_EXTERNAL_KMS"},
+	]
+
+	# Add name to the failed object so that
+	# it fit the format of the alert object
+	failed_obj := json.patch(config_file_content, [{
+		"op": "add",
+		"path": "name",

```

Comment:
Don't we need a metadata property?
I think there was a fix for that https://github.com/kubescape/regolibrary/pull/136

Reply:
we need the "name" property, which is either in the obj level or under metadata